### PR TITLE
feat(granitemoe*): Remove logits upcast when computing loss

### PR DESCRIPTION
**Pull request by gabe-l-hart**
# What does this PR do?
When the CausalLM loss is used, the upcast is done in the loss function utils, so this is redundant.

Fixes #42709 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Review suggestions: @ArthurZucker @Rocketknight1 @Ssukriti

### CODE REVIEW

**Review by Ssukriti — APPROVED** [review]
LGTM , thank you

**Review by ArthurZucker — APPROVED** [review]
Yep ty
